### ISSUE TITLE
knative already support initcontainer，can storage be initializer without webhook

### ISSUE THREAD

**Issue by homily707** [kserve/kserve#3135]
/kind feature
I see comment in [storage_initializer_injector.go](https://github.com/kserve/kserve/blob/master/pkg/webhook/admission/pod/storage_initializer_injector.go#L111) says "This is a work around because KNative does not support INIT containers"
but knative support initContainer now , is that mean we can add initcontainer directly rather than using webhook ?

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Comment by yuzisun** [comment]
@homily707 yes we can add the init container in the kserve controller instead of using the pod mutator. This requires a bit refactoring, welcome any contributions! Cc @greenmoon55

**Comment by homily707** [comment]
I'm glad to help!

**Comment by rhuss** [comment]
This actually makes also sense since it would make it faster to deploy and scale up Knative Pods because you can save the extra webhook (if the storage initializer is the only thing that happens in the mutating webhook).

There is one little caveat, that for #3110 the field `shareProcessNamespace` needs to be set, which is also not yet supported by a Knative service. See https://github.com/knative/serving/issues/14137. I have a fix for that, but it would require a new version of Knative.

Also note, that init-container support needs to be enabled in the Knative installation's config map, by default it is disabled afaik. This might require some documentation update and also possibly issue for backwards compatibility (i.e. existing installations must enable that flag before updating to a new version of Kserve)

**Comment by homily707** [comment]
Maybe we should wait until knative fix that shareProcessNamespace issue, but coding work can begin first. And will customized storage initializer StorageContainer introduce more fields that knative doesn't support? if so, we should keep the webhook.

**Comment by rhuss** [comment]
The PR for Knative Serving to introduce `shareProcessNamespace` is here now --> https://github.com/knative/serving/pull/14454

**Comment by anencore94** [comment]
Hi, any updates on this issue? Can I take this? @homily707 @yuzisun @rhuss

**Comment by rhuss** [comment]
In the meantime, https://github.com/knative/serving/issues/14454 has been merged and is part of the latest Knative releases. Technically, I think all dependencies are unblocked, so we could be good to go. But I leave the final say to @yuzisun as this is a major update, also controlplane-wise (e.g. a deployment less), so a wider consensus should be achieved  ....

**Comment by anencore94** [comment]
@yuzisun Is there any plans for this issue?
